### PR TITLE
Theedge 3683 fix select all

### PR DESCRIPTION
**Pull request by acosferreira**
# Description

For inventory list, implement the select all functionality.
Applying filters and pagination.
Fix enable/disable
Fix count

[THEEDGE-3683.webm](https://github.com/RedHatInsights/edge-frontend/assets/5039367/98a34813-e8b9-45a0-aefd-93590813cc55)

Fixes # (THEEDGE-3683)

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I run `npm run lint:js:fix` to check that my code is properly formatted

### CODE REVIEW

File: src/api/devices/index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { EDGE_API, getTableParams } from '../index';
 import { instance } from '@redhat-cloud-services/frontend-components-utilities/interceptors/interceptors';
 
 export const getInventory = ({ query }) => {
+  console.log(query);
```
Comment (de1987) [inline src/api/devices/index.js:5]: should this be removed?

File: src/components/general-table/BulkSelect.js [hunk 2]
```diff
@@ -7,19 +7,58 @@ import {
...
 }) => {
-  const isAllSelected = checkedRows.length === displayedRowsLength;
-  const isPartiallySelected = checkedRows.length > 0 ? null : false;
+  const isAllSelected = checkedRows.length === total;
+  const isPartiallySelected = checkedRows.length > 0 ? true : false;
   const [selectAllToggle, setSelectAllToggle] = useState(false);
+  const { fetchBatched } = useFetchBatched();
+
+  const fetchAllSystemIds = useCallback((filters, total) => {
+    console.log(filters);
```
Comment (de1987) [inline src/components/general-table/BulkSelect.js:31]: same here

File: src/hooks/useFetchBatched.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import usePromiseQueue from './usePromiseQueue';
+
+const useFetchBatched = () => {
+  const { isResolving: isLoading, resolve } = usePromiseQueue();
+
+  return {
+    isLoading,
+    fetchBatched: (fetchFunction, total, filter, batchSize = 50) => {
+      const pages = Math.ceil(total / batchSize) || 1;
+      console.log(filter);
```
Comment (de1987) [inline src/hooks/useFetchBatched.js:10]: same here

File: src/components/general-table/BulkSelect.js [hunk 4]
```diff
@@ -7,19 +7,58 @@ import {
...
+    const query = apiFilterSort
+      ? {
+          ...filterParams(filters.filters),
+        }
+      : null;
+    return fetchBatched(getInventory, total, query);
```
Comment (de1987) [inline src/components/general-table/BulkSelect.js:37]: just to make sure I understood correctly
will this make multiple requests page by page to the edge-api to retrieve all the ids?

Comment (acosferreira) [reply src/components/general-table/BulkSelect.js:36]: explains during daily

**Review by de1987 — APPROVED** [review]

**Review by ldjebran — APPROVED** [review]
lgtm

**Pull request merged**